content/docs/2.13/authentication-providers/aws.md

```diff
@@ -0,0 +1,141 @@
++++
+title = "Pod Identity Webhook for AWS by using IAM Roles for Service Accounts (IRSA)"
++++
+
+[**AWS IAM Roles for Service Accounts (IRSA) Pod Identity Webhook**](https://github.com/aws/amazon-eks-pod-identity-webhook) ([details](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/)) allows you to provide the role name using an annotation on a service account associated with your pod.
+
+You can tell KEDA to use AWS Pod Identity Webhook via `podIdentity.provider`.
+
+```yaml
+podIdentity:
+  provider: aws
+  roleArn: <role-arn> # Optional. 
+  identityOwner: keda|workload # Optional. If not set, 'keda' is default value. Mutually exclusive with 'roleArn' (if set)
```

Comment:
Is this for non-role ARN scenarios (and thus pod identity?)? Based on the doc it is not clear so maybe we need a "this has 2 auth options and this is how they are different" section

Reply:
roleARN is like clientId in Azure, so KEDA always uses roleArn, but there are 3 different options for providing it:
- Not setting `roleArn` nor `identityOwner` or `identityOwner: keda`: KEDA will use its own role
- Setting `roleArn`: KEDA will use provided roleArn
- Setting `identityOwner: workload`: KEDA will inspect workload service account for getting its role and will use it

Maybe, instead of `identityOwner: workload` we can just use something like `useWorkloadIdentity` and that's all. Maybe it's more clear. My afraid is that `useWorkloadIdentity` can be even more confusing, as WorkloadIdentity has its own meaning in Azure and GCP. 

Currently, we don't support this feature (`identityOwner`) in azure pod identity, but I think that it's something interesting that we can support too

Reply:
Yeah, we talked about adding a new field instead of using roleArn field for both. We suggested 'type', but that can be confusing too... ( because we talked about using it as enum too, like this)
That's why I used identityOwner as it's the current name in AWS scalers. It's agnóstico from the vendor, it already in use for aws, and can be exaplained for other vendors

Reply:
Theoretically, can there be a different `identityOwner` than `keda` or `workload`? If not, then mabye the boolean is a better option?
I'd be okay with both options though.

Reply:
No, it can be `keda` or `workload` (and values are validated by OpenAPI schema). My problem here is how to name it in a clear and concise way because the "concept" of identity is attached to vendor, so useWorkloadIdentity can mean different things on different vendors. As AWS uses "roles", workload identity is quite clear, KEDA'll use the workload role, but for Azure or GCP where Workload Identity is the authentication way itself, this can be more confusing... Am I using Workload Identity (the authentication way) or the workload identity (retrieved using Workload identity authentication way). I'm open to suggestions with the naming and if we find a good naming for a boolean value, I'm more than happy changing it

Reply:
We should also take into account that by omission, KEDA has to be the default value, so useKedaIdentity sounds weird when I have to explicitly set it to false if I want to use podIdentity.roleArn (and podIdentity.clientId in the future) because they are mutually exclusive... 
Maybe something like `useScaledWorkloadIdentity` can be clear enough, but it's a huge name for a key...

Reply:
> As AWS uses "roles", workload identity is quite clear, KEDA'll use the workload role, but for Azure or GCP where Workload Identity is the authentication way itself, this can be more confusing... Am I using Workload Identity (the authentication way) or the workload identity (retrieved using Workload identity authentication way).

I honestly don't know if we should go to a common thing at the cost of less "optimized UX" per cloud provider. The majority of users will only use one cloud, so I'd prefer to make it dead-simple to configure auth for that cloud.

If the approach per cloud provider is different per provider, then it's up to us to clearly help in docs (or separate section) and clarify that.

> Maybe something like `useScaledWorkloadIdentity` can be clear enough, but it's a huge name for a key...

Size does not matter, but booleans limit flexibility in the future. I think the current `identityOwner` is OK and fairly clear.

Reply:
> For KEDA v3, I think that we should refactor this to nested structs for each auth (IMHO it will improve user experience), but until that, I think that less fields are better if we can summarize and reuse them.

I don't know if that is a good way if it means we lose cloud-specific naming/concepts; end-users don't care about code re-use but do care about simplicity.

Reply:
I'm OK with the current approach, however, `identityOwner: workload` makes me expect it uses something from the workload but mostlikely it is the role ARN in the trigger spec? If so, this is confusing IMO and should be `autoscaling` or `autoscalingRule` or so imo

Reply:
`identityOwner: workload` means that KEDA will use the workload roleArn, inspecting the workload's service account looking for the roleArn annotation on it (not in trigger spec). 

> If so, this is confusing IMO and should be autoscaling or autoscalingRule or so imo

I don't get you, sorry :(

Reply:
This is OK, :shipit: